### PR TITLE
feat(contribution): install fingerprint generator — private leak patterns out of tracked source (1/2)

### PR DESCRIPTION
**Pull request by WingedGuardian**
## Why

Leak-detection patterns are DATA with a privacy classification, not neutral code. Today several tracked scanners hardcode this install's specific private literals (hostnames, subnets, ULA prefixes, private repo name, timezone) as their patterns — so the scanners themselves publish what they exist to redact, and the CI scan stays green only because it excludes its own definition files. The fix is a three-layer design: the public tree ships only generic **CLASS** patterns; an install's **specific** literals live out-of-band. This PR is the additive first half — the generator + wiring + tests. **PR 2** genericizes the public scanner surfaces (`sanitize.py`, `.gitleaks.toml`, `ci.yml`, `commit-msg`).

## What this adds

`genesis.contribution.fingerprints` generates `~/.genesis/release-fingerprints.txt` — the file `sanitize._check_fingerprints` already reads (and the pre-push/commit-msg hooks consume) — from local config/environment, so no private byte lives in tracked source.

- **`harvest()`** derives patterns from `genesis.yaml` (service subnets, timezone, `user/private_repo`), the guardian/ambient remote yamls (`host_user`), live `ip -6 addr` ULA prefixes, `$HOME`, the CC project slug, and the hostname. Every source is independently guarded (a fresh clone with none of these produces a valid header-only file and never aborts bootstrap). A **stopword + minimum-specificity guard** ensures generic tokens (e.g. the default `ubuntu` user, a `genesis` hostname) never become BLOCK patterns, and the bare GitHub user (which appears in the public org name) is never emitted — only the `user/private_repo` combination.
- **ERE contract:** emits only `re.escape`'d literals + word boundaries — never `\d`/`\w`/`\s` (a GNU-grep extension that silently mismatches under some `grep -E` builds), so the file is consumable by both Python `re` and `grep -E` (PR 2's commit-msg hook).
- **Marker-delimited** auto-generated block; the hand-edited section (for literals not derivable from config) is preserved verbatim across regeneration. Atomic write at mode 0600 (reuses `util/atomic`).
- **Wiring:** `bootstrap.sh` + `setup-local-config.sh` (non-fatal). Secret-sync — pushing the patterns to the public repo's `GENESIS_PRIVATE_PATTERNS` Actions secret, which PR 2's CI private-scan will read — is **opt-in** via `GENESIS_SYNC_PRIVATE_PATTERNS=1`, resolves the target repo live via `gh` (cwd pinned to the repo root, cross-checked against the configured `public_repo` so it can't push to an unexpected repo), and is inert until PR 2 adds the consumer. Added the file to `backup.sh` (its hand-edited section is not regenerable).

## Activation, not no-op

The fingerprint file does not exist on a current install, so the scanner is inert today. Materializing it **activates** the existing fingerprint scanner in the contribution CLI (hard block) and the pre-push advisory. That is intended and bounded to those user-initiated paths; the specificity guard is the defense against an over-broad harvested pattern.

## Tests

Unit coverage (harvest / source-skipping / marker-preservation / atomic+mode / ERE contract / round-trip through the real `scan_diff` / `sync_secret` success+skip paths / `render` edge cases) plus a **local-only parity proof** (`test_fingerprint_parity.py`) that skips on CI and, on an install with a materialized file, proves each generated pattern blocks its source value — with **no private literal anywhere in the test source** (probes are derived from live-harvested values at runtime).

## Review

genesis-architect pre-build (DONE_WITH_CONCERNS — all findings folded in) → feature-dev code-reviewer (1 BLOCKER on `sync_secret` cwd-dependent repo targeting → fixed with cwd-pinning + configured-repo cross-check + tests; 2 test-gap SHOULD-FIXes → added). Codex is the sequential second reviewer on this PR.

## Deploy path

Runtime code + a new module: `git pull` + bootstrap regen (update.sh re-runs bootstrap). Fresh clone: bootstrap generates a valid file from whatever local config exists. No schema, no new store (the fingerprint file already existed as a read target).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by WingedGuardian — COMMENTED** [review]
## First-pass code review — PR #1272

### Structural review

**Import hygiene — WARNING**

`src/genesis/contribution/fingerprints.py` imports two stdlib modules inside function bodies instead of at the module top level:

- `import os` inside `_default_path()` (line ~105)
- `import re` inside `_escape()` (line ~120)

Both are stdlib and have no reason to be lazy-imported. Python caches them after first import so there's no correctness bug, but it violates the standard import grouping convention and obscures the module's dependency surface. Move both to the top-level stdlib block alongside `argparse`, `contextlib`, etc.

`import yaml` inside `_load_yaml()` is intentionally lazy (third-party, best-effort guard) — that one is fine.

**Missing docstring — NOTE**

`main()` in `fingerprints.py` has no docstring. Minor, but it's a public CLI entry point.

---

### PII / Secrets / Wording scan

**Code diff — CLEAN**

Scanned all added lines. Test fixtures use clearly synthetic values (`AcmeOrg`, `zorrik`, `10.5.6.7`, `fd12:3456`, etc.) — no real private data. No hardcoded API keys, tokens, passwords, SSH material, private IPs in production code, personal email addresses, or connection strings found.

**PR body — SOFT WARNING**

The PR body states: *"Today several tracked scanners hardcode this install's specific private literals (hostnames, subnets, ULA prefixes, private repo name, timezone) as their patterns."* This confirms at a structural level that private literals currently exist in the files PR 2 will modify (`sanitize.py`, `.gitleaks.toml`, `ci.yml`, `commit-msg`). No actual private values are named here, which is correct.

**Action for PR 2:** The reviewer of PR 2 should verify that the diff removing those hardcoded private literals does not inadvertently expose any actual private value (hostname, subnet, ULA prefix, repo name) in the diff itself — the removal diff will contain the old literal as a `-` line visible in git history.

---

### Summary

| Domain | Severity | Finding |
|--------|----------|---------|
| Structural | WARNING | `os` and `re` imported inside function bodies — move to module top level |
| Structural | NOTE | `main()` missing docstring |
| PII/Secrets | SOFT WARNING | PR 2 diff should be reviewed to ensure removed private literals don't persist in git history |

No structural ERRORs. No PII hard blocks. No wording CRITICAL issues.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/genesis/contribution/fingerprints.py [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+        repo_root = Path(__file__).resolve().parents[3]
+
+    out: list[tuple[str, str]] = []
+
+    # 1. genesis.yaml — service subnets, timezone, private repo name.
+    cfg = _load_yaml(home / ".genesis" / "config" / "genesis.yaml")
+    try:
+        net = cfg.get("network", {}) if isinstance(cfg, dict) else {}
+        for key in ("ollama_url", "lm_studio_url"):
+            pat = _ip_prefix_pattern(_host_from_url(str(net.get(key, "") or "")))
```
[the remark this answers is not in the document]
Comment (WingedGuardian) [reply src/genesis/contribution/fingerprints.py]: Fixed in ef610d4c. harvest() now applies env-override > yaml precedence for OLLAMA_URL / LM_STUDIO_URL / USER_TIMEZONE (read from os.environ directly, mirroring the genesis.env precedence, so the generator stays test-injectable and decoupled from the config cache). Added a precedence test. Noted in-code that secrets.env-sourced overrides are seen only if bootstrap exported them into the process.

File: src/genesis/contribution/fingerprints.py [hunk 2]
```diff
@@ -0,0 +1,484 @@
...
+        tz = str(cfg.get("timezone", "") or "").strip()
+        if tz and tz.upper() != "UTC":
+            out.append((r"\b" + _escape(tz) + r"\b", "install timezone"))
+        gh = cfg.get("github", {}) if isinstance(cfg, dict) else {}
+        user = str(gh.get("user", "") or "").strip()
+        priv = str(gh.get("private_repo", "") or "").strip()
+        # Only the user/repo COMBINATION — never the bare user (it appears in
+        # the public org name and would mass-false-positive).
+        if user and priv:
+            out.append((r"\b" + _escape(f"{user}/{priv}") + r"\b", "private repo name"))
```
[the remark this answers is not in the document]
Comment (WingedGuardian) [reply src/genesis/contribution/fingerprints.py]: Fixed in ef610d4c. New _bounded() adds a word boundary only at edges whose char is a word char, so a literal ending in punctuation (e.g. a repo name ending in '-') no longer gets an unmatchable trailing \b. Applied to the repo-name/timezone/host_user/hostname patterns; added a round-trip test proving a punctuation-terminated repo name is still blocked.
